File: Assets/zfoolua/common/Pong.lua

local Pong = {}

function Pong:new()
    local obj = {
        time = 0 -- long
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function Pong:protocolId()
    return 104
end

function Pong:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeLong(packet.time)
end

function Pong:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = Pong:new()
    local result0 = buffer:readLong()
    packet.time = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return Pong

File: Assets/zfoocs/Common/Pong.cs
using System;
using System.Collections.Generic;
namespace zfoocs
{
    
    public class Pong
    {
        public long time;
    }

    public class PongRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 104;
        }
    
        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            Pong message = (Pong) packet;
            buffer.WriteInt(-1);
            buffer.WriteLong(message.time);
        }
    
        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.GetReadOffset();
            Pong packet = new Pong();
            long result0 = buffer.ReadLong();
            packet.time = result0;
            if (length > 0)
            {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}